internal/serialization/default_serializer.go

```diff
@@ -482,8 +501,17 @@ func (JavaArrayListSerializer) Read(input serialization.DataInput) interface{} {
 	return res
 }
 
-func (JavaArrayListSerializer) Write(output serialization.DataOutput, i interface{}) {
-	// no-op
+func (JavaArrayListSerializer) Write(o serialization.DataOutput, i interface{}) {
+	v := i.([]interface{})
+	if v == nil {
+		o.WriteInt32(nilArrayLength)
+		return
+	}
```

Comment:
Can v be nil? If it is nil, it will be serialized with null serializer isn't it? We don't have such logic in ArrayListSerializer in Java side.